### PR TITLE
Add Windows support for gradle itest

### PR DESCRIPTION
**Pull request by BaurzhanSakhariev**
## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: .gitattributes
Comment (amotl) [inline .gitattributes:4]: The documentation for this says:

> When using `text eol=lf`, Git will always convert line endings to `LF` on checkout. You should use this for files that must keep `LF` endings, even on Windows. [1]

So, I believe this is the right decision. Thank you.

[1] https://docs.github.com/en/github/getting-started-with-github/configuring-git-to-handle-line-endings#per-repository-settings

File: devs/docs/tests.rst
Comment (amotl) [inline devs/docs/tests.rst:13]: For properly having links within reStructureText, use `<reference>`_. Also, please use a maximum line length of 80 characters.

```suggestion
For running itest on Windows, `WSL needs to be installed`_. If downloaded Linux
distro comes without Java, install it by running the following commands::
```

File: devs/docs/tests.rst
Comment (amotl) [inline devs/docs/tests.rst:23]:
```suggestion
To support symlinks, `enable Developer Mode`_ and run::
```

File: devs/docs/tests.rst
Comment (amotl) [inline devs/docs/tests.rst:27]: Can you elaborate how to exactly launch WSL from the project directory?

**Review by amotl — CHANGES_REQUESTED** [review]
Hi Baurzhan,

thanks a stack for working on this. I just added some comments.

With kind regards,
Andreas.

File: .gitattributes
Comment (BaurzhanSakhariev) [reply .gitattributes:4]: Hi Andreas, thanks for reviewing! 
I'm using LF so that CR doesn't prevent running scripts which are meant to be run in WSL.
Regarding gradlew - it's even though it's supposed to be used not on WSL but on Windows as well it works without CR - and we need LF here to be able to run `gradlew itest'  from WSL.

File: devs/docs/tests.rst
Comment (BaurzhanSakhariev) [reply devs/docs/tests.rst]: Thanks for pointing it out, good catch.

File: devs/docs/tests.rst
Comment (BaurzhanSakhariev) [reply devs/docs/tests.rst]: Done

File: devs/docs/tests.rst
Comment (amotl) [reply devs/docs/tests.rst:70]: Thanks for spotting this, I overlooked it.

Comment (BaurzhanSakhariev) [reply devs/docs/tests.rst:70]: Done

File: devs/docs/tests.rst
Comment (amotl) [inline devs/docs/tests.rst:70]: Can we also link to https://docs.microsoft.com/en-us/windows/apps/get-started/enable-your-device-for-development here?

File: devs/docs/tests.rst
Comment (amotl) [inline devs/docs/tests.rst:24]:
```suggestion
To support symlinks, `enable Developer Mode`_ on the "For developers" settings
page and run::
```

**Review by amotl — CHANGES_REQUESTED** [review]

File: devs/docs/tests.rst
Comment (BaurzhanSakhariev) [reply devs/docs/tests.rst]: Done. Makes more sense to use official source, thanks.

File: devs/docs/tests.rst
Comment (norosa) [inline devs/docs/tests.rst:14]:
```suggestion
For running itest on Windows, `WSL needs to be installed`_. If the downloaded
Linux distro comes without Java, you can install it by running the following
commands::
```

nit

File: devs/docs/tests.rst
Comment (amotl) [inline devs/docs/tests.rst:19]: It looks like a trailing dot slipped in here.

File: devs/docs/tests.rst
Comment (amotl) [inline devs/docs/tests.rst:28]: While this part of the documentation can stay here, I wonder if we should additionally add a `.. NOTE::` section about "development on Windows" near the `git clone` command within the "Basic setup" document [1].

Regarding this, I assume that the "developer mode" has to be activated before actually cloning the repository. Is this true?

[1] https://github.com/crate/crate/blob/4.5.0/devs/docs/basics.rst#prerequisites

**Review by amotl — COMMENTED** [review]
Hi Baurzhan,

thank you for bringing in all suggestions. I added two more comments.

With kind regards,
Andreas.

File: devs/docs/tests.rst
Comment (amotl) [reply devs/docs/tests.rst:28]: Or, alternatively, let's keep this fenced to this document but add a respective note here that ``git clone`` would have to be invoked again after adjusting the WSL environment.

Comment (norosa) [reply devs/docs/tests.rst:28]: minor nit: GitHub RST rendering doesn't support admonitions (e.g., `NOTE`). check out https://github.com/crate/crate-docs/blob/main/style/hypertext.rst for an example of how I work around this (the note block starting with "Link text noun phrases should not be self-referential.")

Comment (amotl) [reply devs/docs/tests.rst:28]: Oh, true. Thank you for the heads-up, Naomi!

File: blackbox/docs
Comment (amotl) [inline blackbox/docs:1]: Was it intended to remove this?

Comment (BaurzhanSakhariev) [reply blackbox/docs:1]: Hi, no, I accidentally deleted it and restored locally. Trying to check whether it's indeed needed to clone again after enabling dev mode or it's proactive.

File: devs/docs/tests.rst
Comment (BaurzhanSakhariev) [reply devs/docs/tests.rst]: Fixed, thanks

File: devs/docs/tests.rst
Comment (BaurzhanSakhariev) [reply devs/docs/tests.rst:27]: Good point, thanks. Currently checking whether symlink created proactively after enabling dev mode.

Comment (BaurzhanSakhariev) [reply devs/docs/tests.rst:27]: So, just enabling dev mode and setting `symlinks = true` is not enough. I wrote a bit different approach to checkout the file (fetch + checkout) as for cloning one needs to remove directory and re-clone again (If I'm not missing something)

Comment (amotl) [reply devs/docs/tests.rst:27]: That was kind of the thing I was expecting. When symlinks are needed to reflect the repository contents, support for them should be enabled _before_ cloning the repository.

Comment (BaurzhanSakhariev) [reply devs/docs/tests.rst:28]: I found out another way - switching branches also helps in that regard. Added this as a short note. 
I can mention enabling Dev mode in prerequisites as well for double safety - wdyt?

Comment (amotl) [reply devs/docs/tests.rst:28]: Thank you. I believe it is good to go.

**Review by amotl — APPROVED** [review]

**Comment by amotl** [comment]
Thank you. Can I ask you to stash all of your commits into a single one?

**Comment by BaurzhanSakhariev** [comment]
> Thank you. Can I ask you to squash all of your commits into a single one?

Done. Thanks for the review! Sorry for not squashing it immediately, was waiting for at least one approval.

**Pull request merged**
